### PR TITLE
Adding new flow for resetting process

### PR DESCRIPTION
**Pull request by JasonHoffmann**
This de-emphasizes the reset button and adds a step to confirm that this step is more clear.

screen shot 2017-01-24 at 3 53 07 pm

### CODE REVIEW

**Comment by JasonHoffmann** [comment]
Thoughts?

**Comment by zachwills** [comment]
@JasonHoffmann IMO the UI change for the button is unnecessary but I like the additional confirm step

File: assets/src/js/batch.jsx [hunk 1]
```diff
@@ -140,6 +142,11 @@ var App = React.createClass( {
 		this.toggleProcessing( true );
 	},
 
+	toggleResetModal : function( active ) {
+		if( active === false || active === true ) {
```
Comment (zachwills) [inline assets/src/js/batch.jsx]: I don't remember if this project has `eslint` config but this should be `if (` per WP coding standards https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#examples-of-good-spacing

Comment (zachwills) [inline assets/src/js/batch.jsx]: Would `active` ever be something besides `false` or `true`?

File: assets/src/js/components/ModalReset.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Modal component.
+ */
+var ModalReset = React.createClass( {
+	/**
+	 * Type Checking
+	 * @type {Object}
+	 */
+	propTypes: {
```
Comment (zachwills) [inline assets/src/js/components/ModalReset.js]: Spacing in here looks mixed / off.

File: assets/src/js/components/ModalReset.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			classes += ' is-open';
+		}
+
+		return (
+			<div className={ classes }>
+				<div className="batch-overlay__inner">
+         <h3>Are you sure you want to reset this process?</h3>
+         <p>Doing so will reset the last run time back to never, and delete all associated post meta and options data.</p>
+
+         <button id="submit" className="button button-primary" onClick={ this.props.resetBatch.bind( null, 1 ) }>Reset</button>
```
Comment (zachwills) [inline assets/src/js/components/ModalReset.js]: What is the purpose of `null, 1` here?

File: assets/main.css [hunk 4]
```diff
@@ -26,6 +26,25 @@
 	margin-top: 0;
 }
 
+.batch-picker .button-secondary {
+	background: none !important;
```
Comment (zachwills) [inline assets/main.css]: Is there an alternative approach that doesn't require overwriting styles with `!important`?

File: assets/src/js/batch.jsx [hunk 5]
```diff
@@ -236,6 +245,12 @@ var App = React.createClass( {
 					batchErrors={ this.state.errors }
 					toggleProcessing={ this.toggleProcessing }
 				/>
+
+				<ModalError
```
Comment (zachwills) [inline assets/src/js/batch.jsx]: Where is `ModalError` coming from?

File: assets/src/js/batch.jsx (hunk 1 above)
Comment (jjeaton) [inline assets/src/js/batch.jsx]: We do have eslint configured. It looks like it's not failing the `gulp` task, and so it doesn't fail the build. cc @JasonHoffmann

File: assets/src/js/batch.jsx (hunk 5 above)
Comment (jjeaton) [inline assets/src/js/batch.jsx]: It's coming from `import ModalError from './components/ModalReset';`

**Comment by jjeaton** [comment]
A UI change is needed.

We could also try something like this:

2017-01-25 at 10 15 am

Anything to demphasize reset against run.

I think we could also simplify the button labels like this:

2017-01-25 at 10 17 am

**Comment by jjeaton** [comment]
I get these warnings in the console when opening the modal:

2017-01-25 at 10 18 am

File: assets/src/js/batch.jsx (hunk 5 above)
Comment (zachwills) [inline assets/src/js/batch.jsx]: Ah ok. IMO, for clarity, we should either rename the file + component to `ModalError` or change the import + usage to `ModalReset` to match the file. The current mix of `ModalError` and `ModalReset` is confusing. Unless I'm just missing something 🤔

**Comment by zachwills** [comment]
> A UI change is needed.

Why do you think it is needed?

**Comment by jjeaton** [comment]
@zachwills the reset button is useful when developing this plugin, but is a potentially destructive action. Its proximity to the run button is the issue, even with secondary styling. It needs to not be so easy to hit reset, as you may end up re-running a migration on data that's already been modified.

**Comment by JasonHoffmann** [comment]
@jjeaton I like the change in labels "Run" and "Reset Batch", but I think when you toss the button over to the right, it floats in this weird negative space, and it's unclear what exactly it's tied to. What about just adding more margin between them?

image

**Comment by jjeaton** [comment]
The labels change I really like. I could even do without any additional margin, as long as the styling is different. If we can style that w/o the `!important` that would be preferred.

File: assets/src/js/batch.jsx (hunk 1 above)
Comment (JasonHoffmann) [inline assets/src/js/batch.jsx]: Nope

**Comment by JasonHoffmann** [comment]
Ok, the issues above have been addressed. This things ready for a full review.

image

File: assets/src/js/batch.jsx [hunk 6]
```diff
@@ -140,6 +142,11 @@ var App = React.createClass( {
 		this.toggleProcessing( true );
 	},
 
+	toggleResetModal : function ( active ) {
+		if ( active === false || active === true ) {
```
Comment (jjeaton) [inline assets/src/js/batch.jsx]: should we switch these to yoda conditionals?

Comment (JasonHoffmann) [inline assets/src/js/batch.jsx]: Oh, good catch. Yea, we should probably keep things consistent. I'll add this to eslint too

File: assets/src/js/batch.jsx [hunk 7]
```diff
@@ -200,7 +209,7 @@ var App = React.createClass( {
 		}
 
 		// If we are currently processing a batch and there are results.
-		if ( sProcessing.remote_data.current_step !== sProcessing.remote_data.total_steps && sProcessing.remote_data.total_num_results !== 0 ) {
+		if ( sProcessing.remote_data.current_step !== sProcessing.remote_data.total_steps && 0 !== sProcessing.remote_data.total_num_results ) {
```
Comment (jjeaton) [inline assets/src/js/batch.jsx]: Just noticed this. `sProcessing` what's the `s` for? I am having Hungarian notation flashbacks

**Pull request merged**